src/async_wrap.cc

```diff
@@ -582,7 +582,7 @@ void AsyncWrap::EmitDestroy(bool from_gc) {
 
   if (!persistent().IsEmpty() && !from_gc) {
     HandleScope handle_scope(env()->isolate());
-    USE(object()->Set(env()->context(), env()->resource_symbol(), object()));
+    USE(object()->Set(env()->context(), env()->owner_symbol(), object()));
   }
```

Comment:
Should we remove this? I think this is overwriting the previous value at the `owner_symbol` index and causing the `stream[kUpdateTimer] is not a function` failure.

Reply:
I have to admit that I’m not exactly sure on the original motivation here, but I assume that the intention here was what you’re describing, i.e. to override any previous value.

I guess you could also just set it to `null` or something like that.

Reply:
If we set it to `null`, wouldn't it mean bringing back the `resource_symbol`? Or, are you suggesting changing the previous value corresponding to the `owner_symbol` key itself? Wouldn't changing the `owner_symbol` value still cause the same error?

Reply:
> Wouldn't changing the owner_symbol value still cause the same error?

I haven’t tried that out myself yet, I don’t know.

Reply:
@addaleax Setting the value to `null` produces the error: `Cannot read property 'destroyed' of null`. I think it is mandatory to set the value to the JS wrapper object as you mentioned.